src/sql/src/plan/statement/ddl.rs

```diff
@@ -857,12 +866,8 @@ pub fn plan_create_source(
     let create_sql = normalize::create_statement(&scx, Statement::CreateSource(stmt))?;
 
     // Allow users to specify a timeline. If they do not, determine a default timeline for the source.
-    let timeline = if let Some(timeline) = legacy_with_options.remove("timeline") {
-        match timeline.into() {
-            Some(Value::String(timeline)) => Timeline::User(timeline),
-            Some(v) => sql_bail!("unsupported timeline value {}", v.to_ast_string()),
-            None => sql_bail!("unsupported timeline value: secret"),
-        }
+    let timeline = if let Some(timeline) = timeline {
+        Timeline::User(timeline)
```

Comment:
this needs to `require_unsafe_mode` as `legacy_options` already require unsafe mode

Reply:
Already does! https://github.com/MaterializeInc/materialize/blob/e121eda2d7cca47530db6887c29e69c5ba28d47f/src/sql/src/plan/statement/ddl.rs#L336

Reply:
oh wait this is covered by the `SAFE_WITH_OPTIONS` check above, ignore me!

Reply:
oh press send on github at the same time! nice!!